### ISSUE TITLE
Client Error when Connecting to Server hosted on AWS

### ISSUE THREAD

**Issue by madmark82** [magefree/mage#1334]
Hi there,

I'm getting a stack trace when trying to connect to a server on my AWS EC2 instance. I've opened up the 17171 port in AWS and when I attempt to connect via my PC at home I get the following error in my client log file, any ideas?

Thanks in advance.

`FATAL 2015-10-22 18:57:20,798                                                                                            =>[SwingWorker-pool-8-thread-4] SessionImpl.connect 
java.net.UnknownHostException: LITTLEMUFFIN
	at java.net.AbstractPlainSocketImpl.connect(Unknown Source)
	at java.net.PlainSocketImpl.connect(Unknown Source)
	at java.net.SocksSocketImpl.connect(Unknown Source)
	at java.net.Socket.connect(Unknown Source)
	at java.net.Socket.connect(Unknown Source)
	at java.net.Socket.<init>(Unknown Source)
	at java.net.Socket.<init>(Unknown Source)
	at org.jboss.remoting.transport.bisocket.BisocketServerInvoker$2.run(BisocketServerInvoker.java:403)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.jboss.remoting.transport.bisocket.BisocketServerInvoker.createControlConnection(BisocketServerInvoker.java:395)
	at org.jboss.remoting.transport.bisocket.BisocketClientInvoker.transport(BisocketClientInvoker.java:459)
	at org.jboss.remoting.MicroRemoteClientInvoker.invoke(MicroRemoteClientInvoker.java:169)
	at org.jboss.remoting.Client.invoke(Client.java:2084)
	at org.jboss.remoting.Client.addCallbackListener(Client.java:2153)
	at org.jboss.remoting.Client.addListener(Client.java:1258)
	at org.jboss.remoting.Client.addListener(Client.java:1144)
	at org.jboss.remoting.Client.addListener(Client.java:1127)
	at mage.remote.SessionImpl.connect(SessionImpl.java:269)
	at mage.remote.SessionImpl.connect(SessionImpl.java:132)
	at mage.client.MageFrame.connect(MageFrame.java:729)
	at mage.client.dialog.ConnectDialog$ConnectTask.doInBackground(ConnectDialog.java:377)
	at mage.client.dialog.ConnectDialog$ConnectTask.doInBackground(ConnectDialog.java:367)
	at javax.swing.SwingWorker$1.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at javax.swing.SwingWorker.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
FATAL 2015-10-22 18:57:53,477 Communication error                                                                        =>[pool-1-thread-1] SessionImpl.handleThrowable 
java.lang.NullPointerException
	at java.util.concurrent.ConcurrentHashMap.hash(Unknown Source)
	at java.util.concurrent.ConcurrentHashMap.get(Unknown Source)
	at mage.server.SessionManager.extendUserSession(SessionManager.java:214)
	at mage.server.MageServerImpl.ping(MageServerImpl.java:350)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.jboss.remoting.transporter.TransporterHandler.invoke(TransporterHandler.java:87)
	at org.jboss.remoting.ServerInvoker.invoke(ServerInvoker.java:967)
	at org.jboss.remoting.transport.socket.ServerThread.completeInvocation(ServerThread.java:798)
	at org.jboss.remoting.transport.socket.ServerThread.processInvocation(ServerThread.java:751)
	at org.jboss.remoting.transport.socket.ServerThread.dorun(ServerThread.java:586)
	at org.jboss.remoting.transport.socket.ServerThread.run(ServerThread.java:234)
	at org.jboss.remoting.MicroRemoteClientInvoker.invoke(MicroRemoteClientInvoker.java:218)
	at org.jboss.remoting.Client.invoke(Client.java:2084)
	at org.jboss.remoting.Client.invoke(Client.java:879)
	at org.jboss.remoting.transporter.TransporterClient.invoke(TransporterClient.java:475)
	at com.sun.proxy.$Proxy8.ping(Unknown Source)
	at mage.remote.SessionImpl.ping(SessionImpl.java:1417)
	at mage.client.MageFrame$2.run(MageFrame.java:260)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.runAndReset(Unknown Source)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(Unknown Source)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)`

**Comment by LevelX2** [comment]
Look here:
http://www.slightlymagic.net/forum/viewtopic.php?f=70&t=15898

**Comment by madmark82** [comment]
Much appreciated! After reading through that article I found that it was my config settings. I still had 0.0.0.0 set instead of my actual IP. I've graduated to this error now:

`FATAL 2015-10-23 14:39:29,599 Failed to start server - 52.88.126.185:17171/?seri
alizationtype=jboss&maxPoolSize=300      =>[main] Main.main
java.io.IOException: SocketServerInvoker[52.88.126.185:17171] error creating Ser
verSocket[52.88.126.185:17171]: Cannot assign requested address: JVM_Bind
        at org.jboss.remoting.transport.socket.SocketServerInvoker.createServerS
ockets(SocketServerInvoker.java:375)
        at org.jboss.remoting.transport.socket.SocketServerInvoker.start(SocketS
erverInvoker.java:251)
        at org.jboss.remoting.transport.bisocket.BisocketServerInvoker.start(Bis
ocketServerInvoker.java:193)
        at org.jboss.remoting.transport.Connector.start(Connector.java:376)
        at org.jboss.remoting.transporter.TransporterServer.start(TransporterSer
ver.java:198)
        at mage.server.Main.main(Main.java:158)`

Which appears to be explained in the article as well. I'm going to go ahead and close this out. Thanks for the assistance.
